### PR TITLE
Add search by phrase, use better parameter names; more unit-tests

### PR DESCRIPTION
**Pull request by algairim**
Signed-off-by: Mykola Mandra <mykola.mandra@cloudsoftcorp.com>

### CODE REVIEW

File: core/src/test/resources/brooklyn/util/core/logbook/file/log-sample.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+2021-05-27T11:36:59,251 - DEBUG 146 o.a.b.c.m.i.LocalManagementContext [qtp158784971-235] Top-level effector invocation: restart[] on BasicApplicationImpl{id=gwpndj09r8, name=Application (gwpndj09r8)}
```
Comment (algairim) [inline core/src/test/resources/brooklyn/util/core/logbook/file/log-sample.txt:1]: This is a realistic log sample

**Comment by algairim** [comment]
Requires https://github.com/apache/brooklyn-ui/pull/240 to be merge in after, since it changes the API.

File: core/src/main/java/org/apache/brooklyn/util/core/logbook/LogBookQueryParams.java [hunk 2]
```diff
@@ -20,12 +20,28 @@
...
+    /** The indicator whether to return logs in reverse order */
     private Boolean reverseOrder;
+
+    /** The log levels: INFO, FATAL, ERROR, DEBUG or WARNING */
     private List<String> levels;
-    private String initTime;
-    private String finalTime;
+
+    /** The date-time to look log items from */
+    private String dateTimeFrom;
```
Comment (duncangrant) [inline core/src/main/java/org/apache/brooklyn/util/core/logbook/LogBookQueryParams.java:38]: why isn't this a date?

Comment (algairim) [reply core/src/main/java/org/apache/brooklyn/util/core/logbook/LogBookQueryParams.java:38]: Not my choice but a valid point. Looks like LocalDateTime is more appropriate, timezone is not required. What do you think @duncangrant ?

Comment (algairim) [reply core/src/main/java/org/apache/brooklyn/util/core/logbook/LogBookQueryParams.java:38]: Calendar input does take timezone into account though.
Screenshot 2021-07-06 at 18 35 50

**Review by duncangrant — APPROVED** [review]
LGTM thanks @algairim

**Pull request merged**